third-party/linshare/src/main/java/org/apache/james/linshare/client/LinshareAPI.java

```diff
@@ -0,0 +1,83 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.linshare.client;
+
+import java.io.File;
+import java.util.List;
+
+import org.apache.james.linshare.LinshareConfiguration;
+import org.apache.james.linshare.LinshareCredential;
+
+import feign.Feign;
+import feign.Headers;
+import feign.Logger;
+import feign.Param;
+import feign.RequestLine;
+import feign.Retryer;
+import feign.auth.BasicAuthRequestInterceptor;
+import feign.form.FormEncoder;
+import feign.jackson.JacksonDecoder;
+import feign.jackson.JacksonEncoder;
+import feign.slf4j.Slf4jLogger;
+
+public interface LinshareAPI {
+
+    static LinshareAPI from(LinshareConfiguration configuration) {
+        LinshareCredential credential = configuration.getCredential();
+        return Feign.builder()
+            .requestInterceptor(new BasicAuthRequestInterceptor(credential.getUsername(), new String(credential.getPassword())))
+            .logger(new Slf4jLogger(LinshareAPI.class))
+            .logLevel(Logger.Level.FULL)
+            .encoder(new JacksonEncoder())
+            .encoder(new FormEncoder(new JacksonEncoder()))
+            .decoder(new JacksonDecoder())
+            .retryer(new Retryer.Default())
```

Comment:
Is that `retryer` really needed ? - rabbitMQ retry API is a bit specific and fails...